core_data_modules/traced_data/io.py

```diff
@@ -439,6 +445,305 @@ def import_coda_to_traced_data_iterable_as_matrix(cls, user, data, key_of_raw, c
             td.append_data(td_matrix_data, Metadata(user, Metadata.get_call_location(), time.time()))
 
 
+class TracedDataCoda2IO(object):
+    @classmethod
+    def add_message_ids(cls, user, data, raw_key, message_id_key):
+        """
+        Appends a message id to each object in the provided iterable of TracedData.
+
+        Message ids are set by computing the SHA of the value at each `raw_key`, so are guaranteed to be stable.
+
+        If the `raw_key` is not found in a TracedData object in the iterable, no message id is assigned.
+
+        :param user: Identifier of the user running this program, for TracedData Metadata.
+        :type user: str
+        :param data: TracedData objects to set message_ids for.
+        :type data: iterable of TracedData
+        :param raw_key: Key in TracedData objects to read the text to generate message ids for from.
+        :type raw_key: str
+        :param message_id_key: Key in TracedData objects to write the message id to.
+        :type message_id_key: str
+        """
+        for td in data:
+            if raw_key in td:
```

Comment:
This is an example of where we might benefit from more logging. I'd like a verbose mode that outputted after running how many messages had raw_key in td, and how many didn't.

Reply:
Core currently doesn't do any logging anywhere. We should probably talk about how to do that properly.